bftengine/src/bftengine/ControlStateManager.cpp

```diff
@@ -14,12 +14,32 @@
 #include "ControlStateManager.hpp"
 namespace bftEngine {
 
-void ControlStateManager::setStopAtNextCheckpoint() {}
+// This method will be called from the command handler. Notice that the command handler does not aware to the number of
+// the checkpoint. Thus, it can only raise a flag to stop.
+void ControlStateManager::setStopAtNextCheckpoint() {
+  std::ostringstream outStream;
+  controlStateMessages::StopAtNextCheckpointMessage msg(true);
+  concord::serialize::Serializable::serialize(outStream, msg);
+  auto data = outStream.str();
+  state_transfer_.saveReservedPage(
```

Comment:
Not related to this PR. Please change state_transfer_ to pointer.

Reply:
It is a reference, why does it matter if this is a pointer or reference?

Reply:
fixed